edge/model/inference/inference.py

```diff
@@ -234,6 +241,10 @@ def load(load_path, train_x, train_y):
             **construction_parameters
         )
         model.load_state_dict(save_dict['state_dict'])
+
+        if load_dataset:
+            model.dataset.load(load_dataset)
```

Comment:
I think we also need to update the GP's internal state after this: ```self._set_gp_data_to_dataset()```

Reply:
I see. So then how about actually just really making these two things separate? `load` only loads the model. And separately, `load_dataset` loads a dataset. This doesn't make `train_` optional though...